y-octo/src/doc/codec/io/reader.rs

```diff
@@ -38,6 +40,10 @@ pub trait CrdtReader {
     fn read_var_string(&mut self) -> JwstCodecResult<String> {
         read_with_cursor(self.get_buffer_mut(), read_var_string)
     }
+    fn read_intern_var_string(&mut self) -> JwstCodecResult<Spur> {
+        let string = read_with_cursor(self.get_buffer_mut(), read_var_string)?;
```

Comment:
I think we should keep codec itself clean from string interner. There must be a string allocate/deallocate when reading from the buffer and it can be done in item factory. 

The decoder will be dropped after reading finished and obviously it doesn't make sense to have to know how the read content accept by the caller

Reply:
And what I decided to used for such optimization is pure `Arc`, because `parentSub` will only be read once and it is always shared between items by `Store::repair`

Reply:
> And what I decided to used for such optimization is pure `Arc`

We still need a string pool to store the read strings so that we can reuse the same string directly when reading duplicate characters.

And the string interner library contains some optimizations for short strings. These optimizations can effectively reduce memory usage.

It will take some time for us to implement them ourselves, for example: https://github.com/Kixiron/lasso/blob/aed4fb5f30f53d43a7f5b0b5b83777ff315fa207/src/arenas/lockfree.rs#L103-L119

Reply:
<img width="613" alt="image" src="https://github.com/y-crdt/y-octo/assets/25152247/ac2a6bad-83ac-416f-b7e2-71b038a0a7b4">

Simply replacing the parent key with smol_str significantly reduces the memory compared to the main branch.

This seems to be a more lightweight improve.